crates/turborepo-repository/src/package_graph/dep_splitter.rs

```diff
@@ -62,17 +78,21 @@ impl<'a> DependencySplitter<'a> {
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 enum WorkspacePackageSpecifier<'a> {
     Alias(&'a str),
-    // Path(&'a str),
+    Path(&'a RelativeUnixPath),
 }
 
 impl<'a> WorkspacePackageSpecifier<'a> {
     fn new(version: &'a str) -> Option<Self> {
         let version = version.strip_prefix("workspace:")?;
-        let (name, version) = version.rsplit_once('@')?;
-        if version == "*" || version == "^" || version == "~" {
-            Some(Self::Alias(name))
-        } else {
-            None
+        match version.rsplit_once('@') {
+            Some((name, "*")) | Some((name, "^")) | Some((name, "~")) => Some(Self::Alias(name)),
+            // No indication of different name for the package
+            // We want to capture specifiers that have type "directory" by npa which boils down to
+            // checking for slashes: https://github.com/pnpm/npm-package-arg/blob/main/npa.js#L79
+            Some(_) | None if version.contains('/') => {
```

Comment:
The code linked above appears to check for backslashes on windows. Should we do that?

Reply:
Luckily in this case pnpm does not generate `workspace:..\windows-path` and expects these to be unix style.

The npa code is used a lot of places, but luckily here we don't have to worry about the "is this path a system or unix style" question.